server/boot/user.js

```diff
@@ -223,9 +229,20 @@ module.exports = function(app) {
             +challenge.challengeType === 4;
         }
 
-        const completedChallenges = profileUser.completedChallenges.filter(
-          ({ name }) => typeof name === 'string'
-        );
+        const completedChallenges = profileUser.completedChallenges
+          .filter(({ name }) => typeof name === 'string')
+          .map(challenge => {
+              const format = 'MMM DD, YYYY';
+              if (challenge.completedDate) {
+                challenge.completedDate = 
+                  moment.tz(challenge.completedDate, timezone).format(format);
+              }
+              if (challenge.lastUpdated) {
+                challenge.lastUpdated = 
+                  moment.tz(challenge.lastUpdated, timezone).format(format);
+              }
+              return challenge;
+          });
```

Comment:
It is good practice in functional programming to create a new object instead of mutating the old challenge object to prevent unwanted side-effects.

```js
             const format = 'MMM DD, YYYY';
             // use ES7 spread operator to spread the props of the challenge object
             // onto a new object
             challenge  = { ...challenge };
              if (challenge.completedDate) {
                challenge.completedDate = 
                  moment.tz(challenge.completedDate, timezone).format(format);
              }
              if (challenge.lastUpdated) {
                challenge.lastUpdated = 
                  moment.tz(challenge.lastUpdated, timezone).format(format);
              }
              return challenge;
```